### ISSUE TITLE
OHIF/Viewers with DICOMcloud not work fine

### ISSUE THREAD

**Issue by wangning08115** [DICOMcloud/DICOMcloud#76]
First very thanks for DICOMcloud,  It's good project.
Now i want use the OHIF/Viewers project to view dicom file.
OHIF/Viewers is use the last version of master: https://github.com/OHIF/Viewers

#1.  I download code of OHIF/Viewers, and run in localhost , default OHIF/Viewers is connect to server: https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE  ,  it is say the local OHIF/Viewers is work fine with DCM4CHEE  server.
#2. Then i change Viewers to connect localhost DICOMcloud server, config like below:
 {
        name: 'dcomcloud',
        wadoUriRoot: 'http://localhost/dcomcloud/WadoUri',
        qidoRoot: 'http://localhost/dcomcloud/qidors',
        wadoRoot: 'http://localhost/dcomcloud/wadors',
        qidoSupportsIncludeField: true,
        imageRendering: 'wadors',
        thumbnailRendering: 'wadors',
        enableStudyLazyLoad: true,
 }

I find 404 Error for StudiesController.cs
Because DICOMcloud not has the route for ：
 http://localhost/dcomcloud/qidors/studies/999.999.2.19960619.163000/series/999.999.2.19960619.163000.1/metadata

And i add [Route("qidors/studies/{studies}/series/{series}/metadata")] , then can return data.
I use DICOMweb-js project ,  to upload a dicom file to DICOMcloud. Then i can see data list in localhsot  OHIF/Viewers, like below image:
![image](https://user-images.githubusercontent.com/3455705/73337392-338fe600-42af-11ea-99ef-a8dbd098704c.png)

I can see the data list of dicom files, so i think my localhost OHIF/Viewers is fine.
But when i click one item to view detail, then no element can display like below:
![image](https://user-images.githubusercontent.com/3455705/73337554-8a95bb00-42af-11ea-8591-a08a9b55db1b.png)

## Expected Behavior
And when use localhost OHIF/Viewers connect to server: https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE , It is fine like below:
![image](https://user-images.githubusercontent.com/3455705/73337602-a305d580-42af-11ea-9b4b-646cdd89de3e.png)

## Actual Behavior

Dicom file cant display images detail.

**Comment by dannyrb** [comment]
Do you see any network errors in your console? What kind of data are the web requests, made by the OHIF Web Client, returning? How do they differ from the DCM4CHEE equivalents?

**Comment by wangning08115** [comment]
I find some different of DCM4CHEE api and DICOMcloud api:

I say i add a new Route for StudiesController.cs file. [Route("qidors/studies/{studies}/series/{series}/metadata")]
public HttpResponseMessage SearchForStudies
But StudiesController.cs api return data not any changed.

The SearchForStudies action return json like below:
http get: http://localhost/dcomcloud/qidors/studies/999.999.2.19960619.163000/series
http response:
[{
  "00020010": {
    "vr": "UI",
    "Value": [
      "1.2.840.10008.1.2.1"
    ]
  },
  "00080060": {
    "vr": "CS",
    "Value": [
      "MR"
    ]
  },
  "0020000D": {
    "vr": "UI",
    "Value": [
      "999.999.2.19960619.163000"
    ]
  },
  "0020000E": {
    "vr": "UI",
    "Value": [
      "999.999.2.19960619.163000.1"
    ]
  },
  "00200011": {
    "vr": "IS",
    "Value": [
      1
    ]
  },
  "00201209": {
    "vr": "IS",
    "Value": [
      1
    ]
  }
}
]

And the DCM4CHEE api action return below data:
http get: https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.2.276.0.50.192168001099.8252157.14547392.4/series
http response like below:
[
  {
    "00080054": {
      "vr": "AE",
      "Value": [
        "DCM4CHEE"
      ]
    },
    "00080056": {
      "vr": "CS",
      "Value": [
        "ONLINE"
      ]
    },
    "00080060": {
      "vr": "CS",
      "Value": [
        "SR"
      ]
    },
    "0008103E": {
      "vr": "LO",
      "Value": [
        "Research Derived series"
      ]
    },
    "00081190": {
      "vr": "UR",
      "Value": [
        "http://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.2.276.0.50.192168001099.8252157.14547392.4/series/2.25.894077408788372154755797011145827332730"
      ]
    },
    "0020000D": {
      "vr": "UI",
      "Value": [
        "1.2.276.0.50.192168001099.8252157.14547392.4"
      ]
    },
    "0020000E": {
      "vr": "UI",
      "Value": [
        "2.25.894077408788372154755797011145827332730"
      ]
    },
    "00200011": {
      "vr": "IS",
      "Value": [
        99
      ]
    },
    "00201209": {
      "vr": "IS",
      "Value": [
        1
      ]
    }
  }
]

The two json response has a different：
DCM4CHEE  api return  key-value of   ["vr": "UR"]  , and data value "http://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.2.276.0.50.192168001099.8252157.14547392.4/series/2.25.894077408788372154755797011145827332730"

But DICOMcloud api return not the url address key-value.

And no other error in chrome console.

**Comment by Zaid-Safadi** [comment]
@wangning08115 I think you are not using the latest version of DICOMcloud. Also, you should NOT be adding this route:

> And i add [Route("qidors/studies/{studies}/series/{series}/metadata")] , then can return data.
> I use DICOMweb-js project , to upload a dicom file to DICOMcloud. Then i can see data list in localhsot OHIF/Viewers, like below image:

In the above, you are asking the QIDO-RS service to return metadata and this is not what it does. This is already implemented in the WADO-RS service. 
https://github.com/DICOMcloud/DICOMcloud/blob/development/DICOMcloud.Wado.WebApi/Controllers/MetadataController.cs#L25-L37
``` cs
        [HttpGet]
        [Route("wadors/studies/{StudyInstanceUID}/metadata")]
        [Route("api/studies/{StudyInstanceUID}/metadata")]
        public HttpResponseMessage GetStudiesMetadata
        (
            [ModelBinder(typeof(RsStudiesRequestModelBinder))]
            IWadoRsStudiesRequest request
        )
        {
            return WadoService.RetrieveStudyMetadata(request);
        }
```
OHIF had some limitation with specifying different root url for each service so I recently updated DICOMcloud to support same base url for all services. For reference, you can check this issue https://github.com/DICOMcloud/DICOMcloud/issues/59

In summary, get latest version of DICOMcloud and configure your OHIF viewer something simialr to this:
``` json
  servers: {
    dicomWeb: [
      {
        name: 'DCM4CHEE',
        wadoUriRoot: 'http://localhost:44301/api/wadouri',
        qidoRoot: 'http://localhost:44301/api',
        wadoRoot: 'http://localhost:44301/api',
        qidoSupportsIncludeField: true,
        imageRendering: 'wadors',
        thumbnailRendering: 'wadors',
        enableStudyLazyLoad: true,
      },
    ],
  },
```

**Comment by wangning08115** [comment]
Hi very thanks for you ,  I use your config and OHIF work fine.
{
        name: 'dcomcloud',
            wadoUriRoot: 'http://localhost/dcomcloud/api/wadouri',
            qidoRoot: 'http://localhost/dcomcloud/api',
            wadoRoot: 'http://localhost/dcomcloud/api',
        qidoSupportsIncludeField: true,
        imageRendering: 'wadors',
        thumbnailRendering: 'wadors',
        enableStudyLazyLoad: true,
 }

Thanks for everyone , i will research OHIF and dcomcloud.  And use my project. 

Below is my run screen, it is fine.
![image](https://user-images.githubusercontent.com/3455705/73413370-6e912880-4346-11ea-8ec2-d6f563e98f1c.png)

**Comment by UmeshKamboj** [comment]
I am getting this response after update these urls,

![image](https://user-images.githubusercontent.com/30896209/134682604-f56a982e-639e-486b-9d82-6263b60af03b.png)
![image](https://user-images.githubusercontent.com/30896209/134682771-1587e7f7-d074-4f33-ad81-f283c342d228.png)

**Comment by ashu231083** [comment]
![image](https://user-images.githubusercontent.com/9497943/189911581-14900b78-752a-49e2-b4ba-845bcc2c4c22.png)

I have download latest DicomCloud code and OhifViewer. 
Study List its working fine but Metadata API return 404 issue. 

Please help me!

**Comment by Zaid-Safadi** [comment]
@ashu231083  this is caused by the OHIF viewer not sending the headers in the proper format. We need to get OHIF updated for this to work properly
